test: add test for bond port revive

cathay4t opened the pull request:
Add test case to simulate the import use case in
https://bugzilla.redhat.com/show_bug.cgi?id=2207690

The IP on VLAN of bond should be assigned back when bond port link
revived.

liangwen12year commented:
The tier1 test `test_vlan_over_bond_reconnect_on_link_revive` failed in CI.

liangwen12year commented on tests/integration/nm/bond_test.py:
```diff
@@ -71,3 +93,116 @@ def test_bond_balance_slb():
+        # desired IPv6 address plus link local address
+        and len(iface_state[Interface.IPV6].get(InterfaceIPv6.ADDRESS, []))
+        == 2
+    )
```
Instead of just matching the address length, is it possible to verify VLAN of bond got reactivated with **original desired**  IPv4 address ?

cathay4t replied:
Sure, let me do that

liangwen12year commented on tests/integration/nm/bond_test.py:
```diff
@@ -71,3 +93,116 @@ def test_bond_balance_slb():
+                }
+            )
+            # Apply the simple configure again is the key reproducer
+            libnmstate.apply(
```
I would assume that this is simulating the `Reapply` call in NM and verifying if autoconnect is unblocked upon finish afterwards.

cathay4t replied:
yes, it is

cathay4t commented:
Apparently the NM main has the fix, let me narrow the xfail

cathay4t committed:
test: add test for bond port revive

Add test case to simulate the import use case in
https://bugzilla.redhat.com/show_bug.cgi?id=2207690

The IP on VLAN of bond should be assigned back when bond port link
revived.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
New patch uploaded.
Only skip the test on CentOS stream 8 as it does not have fix yet. All EL9 has NM fixed.

Test case updated to check the IP address restored to VLAN interface.
